### PR TITLE
Earl Grey

### PR DESCRIPTION
**Pull request by MadcapJake**
Add Earl Grey support

### CODE REVIEW

**Comment by bucaran** [comment]
Projects using `earlgray` will also need `earlgray-runtime` in the `node_modules/` to compile properly.

```js
'.eg': ['earlgrey/register', 'earlgrey'],
```

**Comment by tkellen** [comment]
@MadcapJake can you please add a supporting test for this in [rechoir](http://github.com/tkellen/js-rechoir)?

**Comment by tkellen** [comment]
Interpret does not support listing two required dependencies in its current form, only fallbacks. If what you are saying is true @bucaran, we will see this break down in the rechoir tests I just requested from @MadcapJake.

**Comment by tkellen** [comment]
PS: Thanks for the PR!

**Comment by bucaran** [comment]
@MadcapJake Oh, boy.

@tkellen Thanks!

**Comment by MadcapJake** [comment]
Added https://github.com/tkellen/js-rechoir/pull/27 and now with the latest version of Earl Grey there is no longer a need for two dependencies, `earlgrey` is enough.

**Comment by bucaran** [comment]
@MadcapJake :+1: Good! Now it's time for some tea.

File: index.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ const extensions = {
   '.coffee': ['coffee-script/register', 'coffee-script'],
   '.coffee.md': ['coffee-script/register', 'coffee-script'],
   '.csv': 'require-csv',
+  '.eg': ['earlgrey/register', 'earlgrey'],
```
Comment (tkellen) [inline index.js]: Will earlgrey register itself automatically in the second case?

Comment (tkellen) [inline index.js]: If not, you'll need to change the second entry to something like this:
https://github.com/tkellen/js-interpret/blob/master/index.js#L15-L19

FWIW, given how new earlgrey is, I'm guessing you don't need this at all. The array of entries is a list of fallbacks for rechoir to use if something fails with any of them. This makes it possible for us to support older versions that may have differing loader mechanisms.

Comment (MadcapJake) [inline index.js]: Ok, thanks! Removed! :+1:

**Comment by tkellen** [comment]
:+1: thanks!

**Pull request merged**